crates/uv-pep508/src/marker/simplify.rs

```diff
@@ -343,20 +343,56 @@ where
     Some(excluded)
 }
 
-/// Returns `Some` if the version expression can be simplified as a star inequality with the given
-/// specifier.
+/// Returns `Some` if the version range can be simplified as a star specifier.
 ///
-/// For example, `python_full_version < '3.8' or python_full_version >= '3.9'` can be simplified to
-/// `python_full_version != '3.8.*'`.
-fn star_range_inequality(range: &Ranges<Version>) -> Option<VersionSpecifier> {
-    let (b1, b2) = range.iter().collect_tuple()?;
-
-    match (b1, b2) {
-        ((Bound::Unbounded, Bound::Excluded(v1)), (Bound::Included(v2), Bound::Unbounded))
-            if v1.release().len() == 2
-                && *v2.release() == [v1.release()[0], v1.release()[1] + 1] =>
-        {
-            Some(VersionSpecifier::not_equals_star_version(v1.clone()))
+/// For positive ranges like `python_full_version >= '3' and python_full_version < '3.1'`,
+/// returns `== '3.0.*'`.
+///
+/// For negative ranges like `python_full_version < '3.8' or python_full_version >= '3.9'`,
+/// returns `!= '3.8.*'`.
+fn star_range_specifier(range: &Ranges<Version>) -> Option<VersionSpecifier> {
+    match range.iter().count() {
+        // Check for positive star range: single segment (Included(v1), Excluded(v2))
+        1 => {
```

Comment:
Could this logic be added to `VersionSpecifier::from_release_only_bounds` instead?